python/lsst/ip/isr/isrFunctions.py

```diff
@@ -233,120 +232,218 @@ def maskITLEdgeBleed(ccdExposure, badAmpDict, itlEdgeBleedSatMinArea=10000,
     badAmpDict : `dict` [`str`, `bool`]
         Dictionary of amplifiers, keyed by name, value is True if
         amplifier is fully masked.
-    itlEdgeBleedSatMinArea : `int`, optional
-        Minimal saturated footprint area where the presence of edge bleeds
-        will be checked.
+    fpCore : `lsst.afw.detection._detection.Footprint`
+        Footprint of saturated core.
     itlEdgeBleedThreshold : `float`, optional
         Threshold above median sky background for edge bleed detection
         (electron units).
     itlEdgeBleedModelConstant : `float`, optional
         Constant in the decaying exponential in the edge bleed masking.
     saturatedMaskName : `str`, optional
         Mask name for saturation.
+    log : `logging.Logger`
+        Logger to handle messages.
     """
-    maskedImage = ccdExposure.maskedImage
-    xmax = maskedImage.image.array.shape[1]
 
-    # Get minimum of amplifier saturation level
-    satLevel = numpy.min([ccdExposure.metadata[f"LSST ISR SATURATION LEVEL {amp.getName()}"]
-                          for amp in ccdExposure.getDetector() if not badAmpDict[amp.getName()]])
+    # Get median of amplifier saturation level
```

Comment:
`log = log if log else logging.getLogger(__name__)`  either here, or when you call the log, to ensure that the warning isn't lost silently.